### PR TITLE
Reduce the number of screen events produced by the automatic screen t…

### PR DESCRIPTION
**Pull request by aalekz**
…racking.

  * Determine which view controller is currently showing.
  * Only queue screen event for the top most view controller (currently visible).

### CODE REVIEW

**Comment by nijikon** [comment]
Refs https://castleio.atlassian.net/browse/CAS-1717

File: Castle/Classes/UIViewController+CASScreen.m [hunk 1]
```diff
@@ -48,15 +49,54 @@ + (void)ca_swizzleViewDidAppear
...
+        UIViewController *root = application.keyWindow.rootViewController;
+        return [self ca_visibleViewController:root];
+    }
+    
+    return nil;
+}
+
++ (UIViewController *)ca_visibleViewController:(UIViewController *)rootViewController
+{
+    // Deterrmine the currently visible controller using rootViewController as a starting point
```
Comment (sebastiansimson) [inline Castle/Classes/UIViewController+CASScreen.m]: Typo "Deterrmine"

File: Example/Castle/DemoViewController.m [hunk 2]
```diff
@@ -0,0 +1,25 @@
+//
+//  DemoViewController.m
+//  Castle_Example
+//
+//  Created by Alexander Simson on 2020-01-10.
+//  Copyright © 2020 Alexander Simson. All rights reserved.
```
Comment (sebastiansimson) [inline Example/Castle/DemoViewController.m]: Copyright info

File: Example/Castle/DemoViewController.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
+//
+//  DemoViewController.h
+//  Castle_Example
+//
+//  Created by Alexander Simson on 2020-01-10.
```
Comment (sebastiansimson) [inline Example/Castle/DemoViewController.h]: Copyright info

**Pull request merged**